File: frontend/src/types/farm.ts
export interface Farm {
  id: number;
  name: string;
  location: string | null;
  description: string | null;
  googleMapsUrl: string | null;
  createdAt: string | null;
  updatedAt: string | null;
}

export interface FarmPayload {
  name: string;
  location?: string | null;
  description?: string | null;
  googleMapsUrl?: string | null;
}

File: nexus-domain/src/main/java/com/nexus/domain/entity/Farm.java
package com.nexus.domain.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "farms")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Farm {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String name;

    private String location;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "google_maps_url", columnDefinition = "TEXT")
    private String googleMapsUrl;

    @Column(name = "system_key", unique = true, length = 100)
    private String systemKey;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @Builder.Default
    @OneToMany(mappedBy = "farm")
    private List<Station> stations = new ArrayList<>();
}
